### PR TITLE
Implemented form validation and submission using yuk and formik library, also using uuid lib to generate unique id

### PR DESCRIPTION
**Pull request by Rohit0301**

### CODE REVIEW

File: src/components/ui/CustomModal.tsx [hunk 1]
```diff
@@ -49,13 +52,18 @@ const CustomModal: FC<Props> = ({
...
+    if (onOk) {
+      onOk(event);
+      handleClose();
+    }
   };
   return (
     <Fragment>
       {buttonComponent ? (
-         React.cloneElement(buttonComponent, { label: buttonText, onClick: handleOpen })
+        React.cloneElement(buttonComponent, {
```
Comment (satishwaghela) [inline src/components/ui/CustomModal.tsx:63]: Is cloning necessary?

File: src/components/ui/CustomModal.tsx [hunk 2]
```diff
@@ -74,7 +82,11 @@ const CustomModal: FC<Props> = ({
               Icon={<CloseOutlined />}
             />
           </Box>
-          {Boolean(modalBody) && modalBody}
+          {modalBody && 
+           React.cloneElement(modalBody, {
+            onClose: handleClose,
```
Comment (satishwaghela) [inline src/components/ui/CustomModal.tsx:87]: Why modalBody needs onClose?

File: src/components/ui/CustomModal.tsx (hunk 1 above)
Comment (Rohit0301) [reply src/components/ui/CustomModal.tsx:63]: Yes, I am passing a buttonComponent like this <CustomButton id="btn" />, and for opening the modal on this button click we need to pass the onClick method that's why I have cloned it.

File: src/components/ui/CustomModal.tsx (hunk 2 above)
Comment (Rohit0301) [reply src/components/ui/CustomModal.tsx:87]: On form submission we need to close the modal thats why, onclose needed in the modalBody

**Pull request merged**